packages/component_code_gen/templates/generate_actions.py

```diff
@@ -1,9 +1,423 @@
+
+## Async options props
+
+The `options` method is an optional method that can be defined on a prop. It is used to dynamically generate the options for a prop and can return a static array of options or a Promise that resolves to an array of options:
+
+```
+[
+  {
+    label: "Human-readable option 1",
+    value: "unique identifier 1",
+  },
+  {
+    label: "Human-readable option 2",
+    value: "unique identifier 2",
+  },
+]
+```
+
+The `label` MUST BE a human-readable name of the option presented to the user in the UI, and the `value` is the value of the prop in the `run` method. The `label` MUST be set to the property that defines the name of the object, and the `value` should be the property that defines the unique identifier of the object.
+
+If an API endpoint exists that can be used to fetch the options for the prop, you MUST define an `async` options method. This allows Pipedream to make an API call to fetch the options for the prop when the user is configuring the component, rather than forcing the user to enter values for the option manually. Think about it: this is so much easier for the user.
+
+Example async options methods:
+
+```
+msg: {
+  type: "string",
+  label: "Message",
+  description: "Select a message to `console.log()`",
+  async options() {
+    // write any node code that returns a string[] (with label/value keys)
+    return ["This is option 1", "This is option 2"];
+  },
+},
+```
+
+```
+board: {
+  type: "string",
+  label: "Board",
+  async options(opts) {
+    const boards = await this.getBoards(this.$auth.oauth_uid);
+    const activeBoards = boards.filter((board) => board.closed === false);
+    return activeBoards.map((board) => {
+      return { label: board.name, value: board.id };
+    });
+  },
+},
+```
+
+```
+async options(opts) {
+  const response = await axios(this, {
+    method: "GET",
+    url: `https://api.spotify.com/v1/me/playlists`,
+    headers: {
+      Authorization: `Bearer \${this.spotify.$auth.oauth_access_token}`,
+    },
+  });
+  return response.items.map((playlist) => {
+    return { label: playlist.name, value: playlist.id };
+  });
+},
+```
+
+## Component Metadata
+
+Registry components require a unique key and version, and a friendly name and description. E.g.
+
+```
+export default {
+  key: "google_drive-list-all-drives",
+  name: "List All Drives",
+  description: "Lists all drives in an account.",
+  version: "0.0.1",
+  type: "action",
+};
+```
+
+Component keys are in the format app_name_slug-slugified-component-name.
+You should come up with a name and a description for the component you are generating.
+In the description, you should include a link to the app docs, if they exist. Or add this as a placeholder: [See docs here]().
+Action keys should use active verbs to describe the action that will occur, (e.g., linear_app-create-issue).
+Always add version "0.0.1" and type "action".
+You MUST add metadata to the component code you generate.
+
+## TypeScript Definitions
```

Comment:
Now that you're generating a lot of different types of components, I'd recommend putting TypeScript defs into a single place and develop some function to return only TypeScript defs relevant to actions, sources, app files, etc. and pass that into these templates